### PR TITLE
Update cdrtools to 3.01a30

### PR DESCRIPTION
**Pull request by fbrosson**

### CODE REVIEW

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 1]
```diff
@@ -0,0 +1,160 @@
+SUMMARY="A highly portable CD/DVD/BluRay command line recording software"
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Please fix the incorrect grammar.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+DESCRIPTION="
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Get rid of the extra newline before the DESCRIPTION actually begins.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+LICENSE="
+	GNU GPL v2
+	CDDL v1
+	"
+SOURCE_URI="http://downloads.sourceforge.net/project/cdrtools/alpha/cdrtools-3.01a30.tar.bz2"
+CHECKSUM_SHA256="5b9a2f98771c9d0097a1e7640727655ece2864eea95f38e5611af2b2f6e6d9cd"
+REVISION="1"
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+
+PATCHES='cdrtools-3.01~a30.patch'
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Use doublequotes.

Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Also, this goes just below the CHECKSUM_SHA256.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+	cmd:cdda2wav = $portVersion compat >= 3
+	cmd:mkhybrid = $portVersion compat >= 3
+	cmd:cdrecord = $portVersion compat >= 3
+	cmd:isodump = $portVersion compat >= 3
+	cmd:isoinfo = $portVersion compat >= 3
+	"
+REQUIRES="
+	haiku
+	"
+BUILD_REQUIRES="
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Add newline before BUILD_REQUIRES.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+	cmd:cdda2ogg = $portVersion compat >= 3
+	cmd:cdda2wav = $portVersion compat >= 3
+	cmd:mkhybrid = $portVersion compat >= 3
+	cmd:cdrecord = $portVersion compat >= 3
+	cmd:isodump = $portVersion compat >= 3
+	cmd:isoinfo = $portVersion compat >= 3
+	"
+REQUIRES="
+	haiku
+	"
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Add newline before BUILD_REQUIRES.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+	cmd:make
+	cmd:sed
+	"
+
+GLOBAL_WRITABLE_FILES="
+	settings/cdrecord keep-old
+	settings/rscsi keep-old
+	"
+
+SOURCE_DIR="cdrtools-3.01"
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: This goes just before (the correct location of) PATCHES.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+		rscsi/rscsi.1 \
+		rscsi/rscsi.c \
+		scgskeleton/scgskeleton.1
+}
+
+BUILD()
+{
+	make GMAKE_NOWARN=true INS_BASE=$prefix INS_RBASE=$prefix LDPATH="" \
+		RUNPATH="" DEFMANBASE=$relativeDocumentationDir
+		# not multi-job safe
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: This goes above the command.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+	devel:libparanoia = $portVersion compat >= 3
+	devel:librscg = $portVersion compat >= 3
+	devel:libscg = $portVersion compat >= 3
+	devel:libsiconv = $portVersion compat >= 3
+	devel:libscgcmd = $portVersion compat >= 3
+	devel:libschily = $portVersion compat >= 3
+	"
+REQUIRES_devel="
+	cdrtools == $portVersion
+	"
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: All of this goes underneath the PROVIDES/REQUIRES, before the BUILD_REQUIRES*.

File: app-cdr/cdrtools/patches/cdrtools-3.01~a30.patch [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
++RUNPATH=	-R$(INS_BASE)/lib
+ 
+ ###########################################################################
+ #
+ # Installation config stuff
+ #
+ ###########################################################################
+-INS_BASE=	/boot/opt/schily
++#INS_BASE=	$(shell finddir B_SYSTEM_DIRECTORY)
```
Comment (waddlesplash) [inline app-cdr/cdrtools/patches/cdrtools-3.01~a30.patch]: No need to change this line if you're just going to comment it out...

**Comment by waddlesplash** [comment]
Please address review comments & squash your commits.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+GLOBAL_WRITABLE_FILES="
+	settings/cdrecord keep-old
+	settings/rscsi keep-old
+	"
+
+SOURCE_DIR="cdrtools-3.01"
+
+PATCHES="cdrtools-3.01~a30.patch"
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Remove newline before PATCHES & SOURCE_DIR, and move them both to just below CHECKSUM.

**Comment by waddlesplash** [comment]
@fbrosson You missed a comment, and there's one you addressed incorrectly so I added another comment.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe (hunk 7 above)
Comment (fbrosson) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Sorry, I forgot to ask the question. Is it the line with the comment "# not multi-job safe" that should be moved 2 lines up (and one tab to the left) ? Thanks.

Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Yes.

Comment (fbrosson) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: I've just made the change, with some extra polishing.
I did a separate commit in order to add cross-build support. I don't know, however, if the Makefiles will support cross-building the way we would expect.
Shall I merge all the commits or should I discard the last one (which tries to add cross-build support) and only merge the other commits?

**Comment by waddlesplash** [comment]
Yeah, discard the last commit & squash the others.

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 11]
```diff
@@ -0,0 +1,166 @@
+SUMMARY="cdrtools is a CD/DVD/Blu-ray premastering and recording software"
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Needs to start with a capital letter.

Comment (fbrosson) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: @waddlesplash: Just fixed.
BTW, should I discard https://github.com/fbrosson/haikuports/commit/a3ee59c1a3b98777d1a4bbd57f7b5c12ef1e4ec8#diff-144005c153d6a7daab3dbabf39a70059 before merging my commits or may merge them all ?

**Comment by fbrosson** [comment]
Well, I guess the recipe is ready, but if you see any extra changes, please let me know.
Thanks @waddlesplash for all your help and feedback!

File: app-cdr/cdrtools/cdrtools-3.01~a30.recipe [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+	rmdir $libDir
+
+	# devel package
+	packageEntries devel \
+		$developDir
+}
+
+# ----- devel package -------------------------------------------------------
+
+SUMMARY_devel="cdrtools development files"
```
Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: Why did these move back to the bottom...?

Comment (fbrosson) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: I browsed several recipes (while looking for templates for cross-building) and noticed that many had the _devel sections at the end. Then I went through your comments to see if you had mentionned such moves but found none... I did recall having dropped a line starting with "# ----- devel package ------", so I came to the *wrong* conclusion that PROVIDES_devel and 
REQUIRES_devel were at the end...
I'll move them back to their first locations, in this order:
PROVIDES
REQUIRES
PROVIDES_devel
REQUIRES_devel
BUILD_REQUIRES
BUILD_PREREQUIRES

Is this order OK, or should I switch the order of REQUIRES and PROVIDES_devel to get this: 

PROVIDES
PROVIDES_devel
REQUIRES
REQUIRES_devel
BUILD_REQUIRES
BUILD_PREREQUIRES

Thanks @waddlesplash

Comment (waddlesplash) [inline app-cdr/cdrtools/cdrtools-3.01~a30.recipe]: The correct order is:

PROVIDES
REQUIRES

PROVIDES_devel
REQUIRES_devel

BUILD_REQUIRES
BUILD_PREREQUIRES

I intend to add a linting task for this, but style issues are a bit hard to detect in Bash...

**Comment by waddlesplash** [comment]
Thank you!

**Pull request merged**